samples/access_token_example/test/access_token.test.js

```diff
@@ -0,0 +1,319 @@
+  const deleteDataGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b/${bucketName}/o/${dataName}`;
+  const deleteDataGcsOptions = {
+    url: deleteDataGcsRequestUrl,
+    method: 'DELETE',
+    headers: headers,
+    'cache-control': 'no-cache',
+  };
+  const deleteDataResponse = await request(deleteDataGcsOptions);
+  assert.strictEqual(deleteDataResponse.status, 200);
+
+  // Delete GCS bucket
+  const deleteGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b/${createResponse.data.name}`;
+  const deleteGcsOptions = {
+    url: deleteGcsRequestUrl,
+    method: 'DELETE',
+    headers: headers,
+    'cache-control': 'no-cache',
+  };
+  const deleteResp = await request(deleteGcsOptions);
+  assert.strictEqual(deleteResp.status, 200);
+});
+
+it('Generate gcp access token, use gcp access token to create pubsub topic, push message to pubsub', async () => {
+  const scope = 'https://www.googleapis.com/auth/pubsub';
+  // generate access token
+  const access_token = await generateGcpToken(
+    region,
+    projectId,
+    registryName,
+    deviceId,
+    scope,
+    'RS256',
+    rsaPrivateKey
+  );
+
+  const headers = {authorization: `Bearer ${access_token}`};
+  // Create pubsub topic
+  const createPubsubRequestUrl = `https://pubsub.googleapis.com/v1/projects/${projectId}/topics/${testTopicName}`;
+  const createPubsubOptions = {
+    url: createPubsubRequestUrl,
+    method: 'PUT',
+    headers: headers,
+    data: {},
+    'content-type': 'application/json',
+    'cache-control': 'no-cache',
+  };
+  const createResponse = await request(createPubsubOptions);
+  assert.strictEqual(createResponse.status, 200);
+
+  // Publish message to pubsub topic
+  const publishPayload = {
+    messages: [
+      {
+        attributes: {
+          test: 'VALUE',
+        },
+        data: Buffer.from('MESSAGE_DATA', 'base64').toString('utf-8'),
+      },
+    ],
+  };
+  const publishPubsubRequestUrl = `https://pubsub.googleapis.com/v1/projects/${projectId}/topics/${testTopicName}:publish`;
+  const publishPubsubOptions = {
+    url: publishPubsubRequestUrl,
+    method: 'POST',
+    headers: headers,
+    data: publishPayload,
+    'content-type': 'application/json',
+    'cache-control': 'no-cache',
+  };
+  const publishResponse = await request(publishPubsubOptions);
+  assert.strictEqual(publishResponse.status, 200);
+
+  // Delete pubsub topic
+  const deletePubsubRequestPath = `https://pubsub.googleapis.com/v1/projects/${projectId}/topics/${testTopicName}`;
+  const deletePubsubOptions = {
+    url: deletePubsubRequestPath,
+    method: 'DELETE',
+    headers: headers,
+    'content-type': 'application/json',
+    'cache-control': 'no-cache',
+  };
+  const deleteResponse = await request(deletePubsubOptions);
+  assert.strictEqual(deleteResponse.status, 200);
+});
+it('Generate gcp access token, exchange ubermint token for service account access token, use service account access token to send cloud iot command', async () => {
```

Comment:
please move the content from this test into access_token.js file with a new region tag `iot_access_token_iot_sendcommand` as a "feature" and add test to that.